contracts/examples/ping-pong-egld/src/lib.rs

```diff
@@ -0,0 +1,179 @@
+#![no_std]
+#![allow(unused_attributes)]
+
+imports!();
+
+mod user_status;
+
+use user_status::UserStatus;
+
+#[elrond_wasm_derive::contract(PingPongImpl)]
+pub trait PingPong {
+	#[init]
+	fn init(
+		&self,
+		fixed_sum: &BigUint,
+		duration: u64,
+		beginning: Option<u64>,
+		max_funds: Option<BigUint>,
+	) {
+		self.set_fixed_sum(fixed_sum);
+		let computed_beginning = beginning.unwrap_or_else(|| self.get_block_nonce());
+		let deadline = computed_beginning + duration;
+		self.set_deadline(deadline);
+		self.set_beginning(computed_beginning);
+		self.set_max_funds(max_funds);
+	}
+
+	#[payable]
+	#[endpoint]
+	fn ping(&self, #[payment] payment: &BigUint) -> SCResult<()> {
+		require!(
+			payment == &self.get_fixed_sum(),
+			"the payment must match the fixed sum"
+		);
+
+		require!(
+			self.get_beginning() <= self.get_block_nonce(),
+			"smart contract not active yet"
+		);
+
+		if let Some(max_funds) = self.get_max_funds() {
+			require!(
+				&self.get_sc_balance() + payment > max_funds,
+				"smart contract full"
+			);
+		}
+
+		let caller = self.get_caller();
+		let user_id = self.get_or_create_user(&caller);
+		let user_status = self.get_user_status(user_id);
+		match user_status {
+			UserStatus::New => {
+				self.set_user_status(user_id, &UserStatus::Registered);
+				Ok(())
+			},
+			UserStatus::Registered => {
+				sc_error!("can only ping once")
+			},
+			UserStatus::Withdrawn => {
+				sc_error!("already withdrawn")
+			},
+		}
+	}
+
+	fn pong_by_user_id(&self, user_id: usize) -> SCResult<()> {
+		let user_status = self.get_user_status(user_id);
+		match user_status {
+			UserStatus::New => {
+				sc_error!("can't pong, never pinged")
+			},
+			UserStatus::Registered => {
+				self.set_user_status(user_id, &UserStatus::Withdrawn);
+				let user_address = self.get_user_address(user_id);
+				self.send_tx(&user_address, &self.get_fixed_sum(), "pong");
```

Comment:
You will need to merge from master, stuff happened since a week ago.
send_tx now asks for &[u8] instead of &str as message, so 
self.send_tx(&user_address, &self.get_fixed_sum(), b"pong");

Reply:
I did a merge, on top of this there were a few other changes I needed to add as well in order for the code to compile: deriving from TypeAbi, adding derive_imports!();